### PR TITLE
Made only non-test package builds produce SRPMs.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

With running regular and test package builds separately we ended up copying the SRPM for the same package twice. There's no need to do that and in scenarios where we only run tests and skip building the RPMs, copying the SRPM to the output would give a false indication, that the RPMs were actually built. With that in mind, I update the logic to skip copying of SRPMs for test package builds.

###### Change Log  

- Updated the logic inside `pkgworker.go` to copy SRPMs only during regular (non-test) package builds.

###### Does this affect the toolchain?  

No.

###### Test Methodology

- PR check.
- Local builds.

### CODE REVIEW

File: toolkit/tools/pkgworker/pkgworker.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/ccachemanager"
+	ccachemanagerpkg "github.com/microsoft/CBL-Mariner/toolkit/tools/internal/ccachemanager"
```
Comment (anphel31) [inline toolkit/tools/pkgworker/pkgworker.go:17]: is this change intentional?

**Review by anphel31 — APPROVED** [review]

File: toolkit/tools/pkgworker/pkgworker.go (hunk 1 above)
Comment (dmcilvaney) [reply toolkit/tools/pkgworker/pkgworker.go:17]: I've seen the same, `go fmt` prefers this. Was going to do the same next time I changed the file.

**Review by dmcilvaney — APPROVED** [review]

File: toolkit/tools/pkgworker/pkgworker.go (hunk 1 above)
Comment (PawelWMS) [reply toolkit/tools/pkgworker/pkgworker.go:17]: Yes and no. The Go plug-in for VS Code adds these for cases where the "basename" of the package from the string doesn't match what's actually the package name where you go to the source. I can revert this line. I'm thinking about updating the package contents themselves to follow what we've been doing with other Go packages but that's a change for `main`.

Comment (PawelWMS) [reply toolkit/tools/pkgworker/pkgworker.go]: Created a separate PR here: #6958.
